src/NServiceBus.Core/Pipeline/Incoming/LoadHandlersConnector.cs

```diff
@@ -57,8 +65,37 @@ async Task<CompletableSynchronizedStorageSession> AdaptOrOpenNewSynchronizedStor
                    ?? await synchronizedStorage.OpenSession(contextBag).ConfigureAwait(false);
         }
 
+        static void LogHandlersInvocation(IIncomingLogicalMessageContext context, List<MessageHandler> handlersToInvoke)
+        {
+            var builder = new StringBuilder();
+            builder.AppendLine($"Processing message type: {context.Message.MessageType}");
+            builder.AppendLine("Message headers: ");
+
+            foreach (var kvp in context.Headers)
+            {
+                builder.Append($"{kvp.Key}:{kvp.Value}, ");
```

Comment:
I'm actually thinking I'd prefer to see this using `AppendLine` instead. Seems like it would be much easier to read if each header entry was on a separate line.

It also removes the need to set adjust the length after the loop.

Reply:
👍
Whatever the decision made, it will be both for `LoadHandlersConnector` and `RoutingToDispatchConnector`.

Reply:
I'm with Brandon. From a consumption standpoint line by line is going to be much easier to grok. I'd also change the ":" delimiter to " : " to make recognizing the separation a bit clearer